### PR TITLE
[+] Search - Display highlighted matches on table view when searching

### PR DESCRIPTION
**Pull request by root-io**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [ ] Create automatic tests
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by root-io** [comment]
[(13) Search - Users want to see highlighted matches on table view when searching](https://trello.com/c/h2GOQgdC/2168-13-search-users-want-to-see-highlighted-matches-on-table-view-when-searching)

File: services/resources-getter.js [hunk 1]
```diff
@@ -154,14 +157,27 @@ function ResourcesGetter(model, opts, params) {
...
         if (schema.isCompositePrimary) {
           records.forEach(function (record) {
             record.forestCompositePrimary =
               new CompositeKeysManager(model, schema, record)
                 .createCompositePrimary();
           });
         }
-        return [count, records];
+
+        return [records, count, decorators];
```
Comment (arnaudbesnier) [inline services/resources-getter.js]: You broke the CSV Exporter by the way.

Comment (root-io) [reply services/resources-getter.js]: Thanks! Fixed

File: services/search-builder.js [hunk 2]
```diff
@@ -14,6 +14,11 @@ function SearchBuilder(model, opts, params, fieldNamesRequested) {
   var hasSearchFields = schema.searchFields && _.isArray(schema.searchFields);
   var searchAssociationFields;
   var OPERATORS = new Operators(opts);
+  var columns = [];
+
+  this.getColumns = function () {
```
Comment (arnaudbesnier) [inline services/search-builder.js]: Rename to be more explicit `getColumnsSearched`, `getFieldsSearched`?

File: utils/records-decorator.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+function decorateForSearch(records, fieldsSearched, searchValue) {
+  if (records[0] && records[0].dataValues) {
+    records = records.map((record) => record.get({ plain: true }));
+  }
```
Comment (arnaudbesnier) [inline utils/records-decorator.js]: You can get rid of those 3 lines and do `record.attributes` to simplify your code

File: services/has-many-getter.js [hunk 4]
```diff
@@ -57,22 +59,37 @@ function HasManyGetter(model, association, opts, params) {
     return findQuery(queryOptions)
       .then(function (records) {
         return P.map(records, function (record) {
-          // NOTICE: Do not use "toJSON" method to prevent issues on models that
-          //         override this method.
-          var recordFormated = record.get({ plain: true });
```
Comment (arnaudbesnier) [inline services/has-many-getter.js]: I remove this, to have consistent input in the `records-decorator`

File: services/has-many-getter.js [hunk 5]
```diff
@@ -57,22 +59,37 @@ function HasManyGetter(model, association, opts, params) {
...
-          return recordFormated;
+          return record;
         });
       });
   }
 
   this.perform = function () {
-    return P.all([count(), getRecords()]);
+    return P.all([getRecords(), getCount()])
+      .then((results) => {
```
Comment (arnaudbesnier) [inline services/has-many-getter.js]: `=>` is not authorized.

File: utils/records-decorator.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+function decorateForSearch(records, fieldsSearched, searchValue) {
+  if (records[0] && records[0].dataValues) {
+    records = records.map((record) => record.get({ plain: true }));
```
Comment (arnaudbesnier) [inline utils/records-decorator.js]: `=>` is not authorized.

File: utils/records-decorator.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  }
+  var matchFields = {};
+  records.forEach((record, index) => {
+    Object.keys(record).forEach(attributeName => {
```
Comment (arnaudbesnier) [inline utils/records-decorator.js]: `=>` is not authorized.

File: utils/records-decorator.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+function decorateForSearch(records, fieldsSearched, searchValue) {
+  if (records[0] && records[0].dataValues) {
+    records = records.map((record) => record.get({ plain: true }));
+  }
+  var matchFields = {};
+  records.forEach((record, index) => {
```
Comment (arnaudbesnier) [inline utils/records-decorator.js]: `=>` is not authorized.

File: utils/records-decorator.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  });
+
+  if (_.isEmpty(matchFields)) {
+    matchFields = null;
+  }
+
+  return matchFields;
+}
+
+exports.decorateForSearch = decorateForSearch;
```
Comment (arnaudbesnier) [inline utils/records-decorator.js]: This service as to be exposed to the `forest-express` liana to make it run just before the serialization.
The problem with the current implementation is that Smart Fields are not highlighted if they contain the search value. And Smart Field values injection is done in the `forest-express` liana...

**Pull request merged**